node/pkg/node/options.go

```diff
@@ -171,11 +173,32 @@ func GuardianOptionStatusServer(statusAddr string) *GuardianOption {
 				// Prometheus metrics (safe to expose to untrusted clients)
 				router.Handle("/metrics", promhttp.Handler())
 
-				go func() {
+				// SECURITY: If making changes, ensure that we always do `router := mux.NewRouter()` before this to avoid accidentally exposing pprof
+				server := &http.Server{
+					Addr:              statusAddr,
+					Handler:           router,
+					ReadHeaderTimeout: time.Second, // SECURITY defense against Slowloris Attack
+					ReadTimeout:       time.Second,
+					WriteTimeout:      time.Second,
+				}
+
+				g.runnables["status-server"] = func(ctx context.Context) error {
+					logger := supervisor.Logger(ctx)
+					go func() {
+						if err := server.ListenAndServe(); err != http.ErrServerClosed {
+							logger.Error("status server crashed", zap.Error(err))
+						}
+					}()
+					time.Sleep(time.Millisecond * 10) // we don't know exactly when the server will be ready to accept connections, but a few ms should be enough time
```

Comment:
Famous last words :)

Reply:
should be fine, nothing depends on this.

Reply:
seems hacky but :shrug: what is the importance of waiting here?

Reply:
I actually just removed this since it's not necessary and probably just adds more confusion.